### ISSUE TITLE
pg_show_plans & logical replication slot

### ISSUE THREAD

**Issue by pmpetit** [cybertec-postgresql/pg_show_plans#14]
Hello, 
after having installed the extension pg_show_plans  (i can reproduce it when you want),
i face an error when creating logical replication slots.
i created a dedicated cluster for that : 

1/ compile pg_show_plans on the server
2/ add shared_preload_libraries: pg_show_plans 
3/ create extension pg_show_plans.

then, on the server : 
postgres@stock(test)=# select * from pg_create_logical_replication_slot('slot1','pgoutput');
ERROR:  cannot create logical replication slot in transaction that has performed writes
postgres@stock(test)=# 

![image](https://user-images.githubusercontent.com/4059185/84119499-b9c5f100-aa34-11ea-894f-12ed13e328a2.png)
where 7910 is my own pid, it is my session.

if i remove the extension, stop/start cluster:
![image](https://user-images.githubusercontent.com/4059185/84119739-16c1a700-aa35-11ea-9919-3931bac6ecf3.png)

shared_preload_libraries is still set to pg_show_plans:
![image](https://user-images.githubusercontent.com/4059185/84119891-525c7100-aa35-11ea-9d90-e0a26bf61e37.png)

if i remove pg_show_plans from shared_preload_libraries
![image](https://user-images.githubusercontent.com/4059185/84119998-7cae2e80-aa35-11ea-9960-712fbf628b0a.png)

then i can create a logical replication slot:
![image](https://user-images.githubusercontent.com/4059185/84120059-93ed1c00-aa35-11ea-8d77-4d1c4b61ab76.png)

I can test more if you want, just ask.

ps : pgsql 12.2
pg_show_plans  1.0
(i have the same problem on pg10 & pg11)

if you want to reproduce the error without pg_show_plans:
https://github.com/postgres/postgres/blob/master/contrib/test_decoding/expected/decoding_in_xact.out

my .psqlrc
[psqlrc.zip](https://github.com/cybertec-postgresql/pg_show_plans/files/4750898/psqlrc.zip)

ps  : for those who are blocked, you can still create replication slot using
```sh
pg_recvlogical
```
but for pg_squeeze no way.

**Comment by pmpetit** [comment]
my .psqlrc

-- Formation Prompt
\set PROMPT1 '%[%033[1;32;40m%]%n@%/(%`basename $PGDATA`)%R%[%033[0m%]%# '
\set PROMPT2 '%[%033[1;32;40m%]%n@%/(%`basename $PGDATA`)%R%[%033[0m%]%# '

-- History by user
\set HISTFILE `echo $HOME`/.psql_history-`who am i 2>/dev/null | awk '{print $1}' 2>/dev/null; exit`
\!  #sudo chattr +a `echo $HOME`/.psql_history-`who am i 2>/dev/null | awk '{print $1}' 2>/dev/null; exit` 2>/dev/null

-- Intelligent fromatting
\x auto

-- Unused indexes
\set unused_indexes 'SELECT schemaname AS schema, relname AS table, indexrelname AS index FROM pg_stat_user_indexes WHERE idx_scan = 0;'
\set unused_indexes_size 'SELECT relname AS table, indexrelname AS index, pg_size_pretty(pg_relation_size(indexrelid)) AS size FROM pg_stat_user_indexes WHERE idx_scan = 0 ORDER BY pg_relation_size(indexrelid) DESC;'

-- Slow Queries
\set slow_queries 'SELECT (total_time / 1000 / 60) as total_minutes, (total_time/calls) as average_time, query FROM pg_stat_statements ORDER BY 1 DESC LIMIT 100;'

-- Autocomplete keywords (like SELECT) in upper-case, even if you started
-- typing them in lower case.
\set COMP_KEYWORD_CASE upper

**Comment by s-hironobu** [comment]
The pg_create_logical_replication_slot() function returns error in default setting  because the CreateInitDecodingContext() function invoked by this function and the GetTopTransactionId() function invoked by this module make a conflict.

To deal with this conflict, I added a parameter pg_show_plans.enable. I show a workaround below.

```
postgres=# SELECT pg_create_logical_replication_slot('myslot', 'pgoutput');
ERROR:  cannot create logical replication slot in transaction that has performed writes

postgres=# set pg_show_plans.enable = off;
SET
postgres=# SELECT pg_create_logical_replication_slot('myslot', 'pgoutput');
 pg_create_logical_replication_slot
------------------------------------
 (myslot,0/1572F90)
(1 row)

postgres=# SET pg_show_plans.enable = on;
SET
```

Although there is another way --- I already implemented for secondary servers in replication mode --- to deal with this conflict, it is inefficient.
Moreover, only the pg_create_logical_replication_slot function makes this conflict.
Therefore,  in order to minimize the impact, I decided to add the parameter pg_show_plans.enable.
